<a id="1"></a>
<div style="padding:20px;
            color:#F0F3FF;
            margin:10;
            font-size:200%;
            text-align:left;
            display:fill;
            border-radius:15px;
            background-color:#211951;
            overflow:hidden;
            font-weight:900">
            <span style='color:#15F5BA'></span>Descriptive Statistics
</div>
<br>Author : <i>Pandu Giswantoro Abhi Wiguno</i>

# <b>A. <span style='color:#3A6D8C'>Introduction</span></b>

Statistika deskriptif adalah cabang statistika yang berfokus pada pengumpulan, penyajian, dan interpretasi data untuk memberikan gambaran atau deskripsi yang ringkas tentang suatu populasi atau sampel. Tujuannya adalah untuk menggambarkan dan meringkas karakteristik utama dari data yang diamati, seperti ukuran pemusatan (mean, median, modus), ukuran penyebaran (range, standar deviasi), distribusi data, dan hubungan antara variabel.<br><br>

<p align="center">
<img src="https://www.investopedia.com/thmb/_9Vu4GkDksKqtM59z8OtE6svNkg=/750x0/filters:no_upscale():max_bytes(150000):strip_icc():format(webp)/Descriptive_statistics-5c8c9cf1d14d4900a0b2c55028c15452.png" width="500">
<p align="center"><b>Gambar 1. Statistika Deskriptif</b></p>
</p>

# <b>B. <span style='color:#3A6D8C'>Data Digunakan</span></b>

Pada tahun 2020, setelah mengumumkan bahwa operasi perusahaan mereka secara resmi telah mencapai netralitas karbon, Apple berjanji untuk menjadikan produk mereka netral karbon pada tahun 2030. Untuk mencapai tujuan ini, mereka menetapkan emisi tahun 2015 (38,4 juta metrik ton CO2e) sebagai dasar dan akan berusaha menguranginya sebesar 75% pada tahun 2030. Sisa 25% dari emisi kotor (9,6 juta metrik ton CO2e) akan dihilangkan menggunakan offset karbon, sehingga membawa emisi neto menjadi 0.

<a href="https://www.apple.com/newsroom/2020/07/apple-commits-to-be-100-percent-carbon-neutral-for-its-supply-chain-and-products-by-2030/"><i>https://www.apple.com/newsroom/2020/07/apple-commits-to-be-100-percent-carbon-neutral-for-its-supply-chain-and-products-by-2030/</i></a>

### <b><span style='color:#55679C'>Quest B.1</span></b>

<i>Lakukan ekstraksi data untuk data `carbon_footprint_by_product.csv`, `greenhouse_gas_emissions.csv` dan `normalizing_factors.csv` serta lakukan inspeksi tipe data dan informasi umum lainnya!)</i>

In [18]:
# Ekstrak data carbon_footprint_by_product
import pandas as pd
# Proses ekstraksi data
carbon_product = pd.read_csv('/content/carbon_footprint_by_product.csv')
# Tampilkan data
display(carbon_product)


Release Year    Product  Baseline Storage  Carbon Footprint
0          2023  iPhone 15               128                56
1          2022  iPhone 14               128                61
2          2021  iPhone 13               128                64
3          2020  iPhone 12                64                70
4          2019  iPhone 11                64                72
5          2018  iPhone Xs                64                70
6          2017   iPhone X                64                79
7          2017   iPhone 8                64                57
8          2016   iPhone 7                32                56
9          2015  iPhone 6s                32                54

In [19]:
# Informasi umum pada dataframe carbon_product
display(carbon_product.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Release Year      10 non-null     int64 
 1   Product           10 non-null     object
 2   Baseline Storage  10 non-null     int64 
 3   Carbon Footprint  10 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 448.0+ bytes


None

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

In [20]:
# Proses ekstraksi data
gas_emissions = pd.read_csv('greenhouse_gas_emissions.csv')

# Tampilkan data
display(gas_emissions)

Fiscal Year                      Category             Type  \
0           2022           Corporate emissions  Gross emissions   
1           2022           Corporate emissions  Gross emissions   
2           2022           Corporate emissions  Gross emissions   
3           2022           Corporate emissions  Gross emissions   
4           2022           Corporate emissions  Gross emissions   
..           ...                           ...              ...   
131         2015  Product life cycle emissions  Gross emissions   
132         2015  Product life cycle emissions  Gross emissions   
133         2015  Product life cycle emissions  Gross emissions   
134         2015  Product life cycle emissions  Gross emissions   
135         2015  Product life cycle emissions  Carbon removals   

                      Scope                                       Description  \
0                   Scope 1                      Natural gas, diesel, propane   
1                   Scope 1                                    Fleet vehicles   
2                   Scope 1         Other (R&D processes & refrigerant leaks)   
3    Scope 2 (market-based)                                       Electricity   
4    Scope 2 (market-based)                       Steam, heating, and cooling   
..                      ...                                               ...   
131                 Scope 3      Manufacturing (purchased goods and services)   
132                 Scope 3  Product transportation (upstream and downstream)   
133                 Scope 3                Product use (use of sold products)   
134                 Scope 3                            End-of-life processing   
135                     NaN                            Product carbon offsets   

      Emissions  
0       39700.0  
1       12600.0  
2        2900.0  
3           0.0  
4        3000.0  
..          ...  
131  29600000.0  
132   1300000.0  
133   6600000.0  
134    500000.0  
135         NaN  

[136 rows x 6 columns]

In [21]:
# Informasi umum pada dataframe gas_emissions
gas_emissions.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 136 entries, 0 to 135
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Fiscal Year  136 non-null    int64  
 1   Category     136 non-null    object 
 2   Type         136 non-null    object 
 3   Scope        120 non-null    object 
 4   Description  136 non-null    object 
 5   Emissions    109 non-null    float64
dtypes: float64(1), int64(1), object(4)
memory usage: 6.5+ KB


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

In [23]:
# Proses ekstraksi data
normalizing_factors = pd.read_csv('normalizing_factors.csv')

# Tampilkan data
display(normalizing_factors)

Fiscal Year  Revenue  Market Capitalization  Employees
0         2022   394328                   2490     164000
1         2021   365817                   2450     154000
2         2020   274515                   1720     147000
3         2019   260174                   1090     137000
4         2018   265595                    830     132000
5         2017   229234                    740     123000
6         2016   215639                    600     116000
7         2015   233715                    580     110000

In [24]:
# Informasi umum pada dataframe normalizing_factors
normalizing_factors.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8 entries, 0 to 7
Data columns (total 4 columns):
 #   Column                 Non-Null Count  Dtype
---  ------                 --------------  -----
 0   Fiscal Year            8 non-null      int64
 1   Revenue                8 non-null      int64
 2   Market Capitalization  8 non-null      int64
 3   Employees              8 non-null      int64
dtypes: int64(4)
memory usage: 384.0 bytes


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

# <b>C. <span style='color:#3A6D8C'>Central Tendency</span></b>

Ukuran pemusatan adalah ukuran statistik yang digunakan untuk menggambarkan pusat atau titik-titik sentral dari sebuah distribusi data. Ukuran pemusatan memberikan informasi tentang nilai "*typical*" atau representatif dari kumpulan data.

Ukuran pemusatan yang umum digunakan adalah :
* Mean (rata-rata)
* Median (nilai tengah)
* Modus (nilai yang paling sering muncul)
* Kuartil (membagi data menjadi 4 bagian)

Pemilihan ukuran pemusatan yang tepat tergantung pada sifat data dan tujuan analisis statistik yang dilakukan.

---
## <b>C.1. <span style='color:#3A6D8C'> Mean ( *X̄* )</span></b>
---

Mean adalah rata-rata dari angka-angka yang diberikan dan dihitung dengan membagi jumlah angka-angka tersebut dengan total jumlah angka.<br><br>

\begin{equation}
\bar{X} =  \frac{Jumlah\;Data}{Banyak\;Data}
\end{equation}

Contoh :      

Gaji 5 orang pegawai IT di PT. X dengan posisi masing-masingnya adalah sebagai berikut :

<table align="center">
  <tr>
    <th>Nomor</th>
    <th align="left">Posisi</th>
    <th>Gaji</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="left">Data Analyst</td>
    <td align="center">5.500.000</td>
  </tr>
  <tr>
    <td align="center">2</td>
    <td align="left">Data Scientist</td>
    <td align="center">7.500.000</td>
  </tr>
  <tr>
    <td align="center">3</td>
    <td align="left">Data Engineer</td>
    <td align="center">8.000.000</td>
  </tr>
  <tr>
    <td align="center">4</td>
    <td align="left">Business Intelligence</td>
    <td align="center">7.000.000</td>
  </tr>
  <tr>
    <td align="center">5</td>
    <td align="left">Quality Assurance</td>
    <td align="center">5.000.000</td>
  </tr>
</table>
<br>

Berapa rata-rata gaji karyawan IT di perusahaan tersebut?

**Jawab :**

\begin{equation}
\bar{X} =  \frac{Jumlah\;Data}{Banyak\;Data} = \frac{5.500.000 + ... + 5.000.000}{5} = Rp.\;6.600.000
\end{equation}
<br>

Jadi rata-rata gaji karyawan IT di PT. X adalah Rp. 6.600.000<br><br>

> **Interpretasinya** : *Misalnya, seseorang akan melamar sebagai karyawan IT di perusahaan tersebut, dengan mengetahui rata-rata gaji karyawan IT di perusahaan itu maka ia akan dapat memperkirakan rata-rata gaji yang akan bisa ia terima.*

<br>
Untuk menghitung mean dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.mean()
```

docs : *https://pandas.pydata.org/docs/reference/api/pandas.Series.mean.html*

In [26]:
# Write your code below
conditions = gas_emissions['Description'] == 'Manufacturing (purchased goods and services)'

manufacture = gas_emissions.loc[conditions]

display(manufacture)

Fiscal Year                      Category             Type    Scope  \
12          2022  Product life cycle emissions  Gross emissions  Scope 3   
29          2021  Product life cycle emissions  Gross emissions  Scope 3   
46          2020  Product life cycle emissions  Gross emissions  Scope 3   
63          2019  Product life cycle emissions  Gross emissions  Scope 3   
80          2018  Product life cycle emissions  Gross emissions  Scope 3   
97          2017  Product life cycle emissions  Gross emissions  Scope 3   
114         2016  Product life cycle emissions  Gross emissions  Scope 3   
131         2015  Product life cycle emissions  Gross emissions  Scope 3   

                                      Description   Emissions  
12   Manufacturing (purchased goods and services)  13400000.0  
29   Manufacturing (purchased goods and services)  16200000.0  
46   Manufacturing (purchased goods and services)  16100000.0  
63   Manufacturing (purchased goods and services)  18900000.0  
80   Manufacturing (purchased goods and services)  18500000.0  
97   Manufacturing (purchased goods and services)  21100000.0  
114  Manufacturing (purchased goods and services)  22800000.0  
131  Manufacturing (purchased goods and services)  29600000.0

### <b><span style='color:#55679C'>Quest C.1</span></b>
<i>Pada dataframe `gas_emissions`, hitung rata-rata emisi yang dihasilkan oleh Manufacturing pada periode data!</i>

In [27]:
# Write your code below
mean = manufacture['Emissions'].mean()

# menampilkan hasil
print('Rata-rata emisi gas yang dihasilkan oleh Manufaktur Apple adalah', mean)

Rata-rata emisi gas yang dihasilkan oleh Manufaktur Apple adalah 19575000.0


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

---
## <b>C.2. <span style='color:#3A6D8C'> Median (Med)</span></b>
---

Median adalah nilai di tengah deret nilai yang disusun dengan data urut dari yang kecil sampai besar. Median memberikan perkiraan nilai tengah atau pusat dari sekelompok data.<br><br>

\begin{equation}
Med =
\begin{cases}
    X_\frac{n+1}{2}, & \text{jika }n\text{ ganjil}  \\
    \frac{1}{2}(X_\frac{n}{2}+X_\frac{n+1}{2}), &  \text{jika }n\text{ genap}
\end{cases}
\end{equation}

Contoh :      

Gaji 5 orang pegawai IT di PT. X dengan posisi masing-masingnya adalah sebagai berikut :

<table align="center">
  <tr>
    <th>Nomor</th>
    <th align="left">Posisi</th>
    <th>Gaji</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="left">Data Analyst</td>
    <td align="center">5.500.000</td>
  </tr>
  <tr>
    <td align="center">2</td>
    <td align="left">Data Scientist</td>
    <td align="center">7.500.000</td>
  </tr>
  <tr>
    <td align="center">3</td>
    <td align="left">Data Engineer</td>
    <td align="center">8.000.000</td>
  </tr>
  <tr>
    <td align="center">4</td>
    <td align="left">Business Intelligence</td>
    <td align="center">7.000.000</td>
  </tr>
  <tr>
    <td align="center">5</td>
    <td align="left">Quality Assurance</td>
    <td align="center">5.000.000</td>
  </tr>
</table>
<br>

Berapa median (nilai tengah) gaji karyawan IT di perusahaan tersebut?

**Jawab :**

Urutkan data gaji terlebih dahulu

<table align="center">
  <tr>
    <th>5.000.000</th>
    <th>5.500.000</th>
    <th><b>7.000.000</b></th>
    <th>7.500.000</th>
    <th>8.000.000</th>
  </tr>
</table>

Maka diperoleh nilai tengahnya adalah Rp. 7.000.000<br><br>

> **Interpretasi** : *Median yang diperoleh akan memberikan perkiraan gaji tengah, di mana setengah dari karyawan memiliki gaji di atas median dan setengahnya lagi memiliki gaji di bawah median.*



Untuk menghitung median dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.median()
```

docs : *https://pandas.pydata.org/docs/reference/api/pandas.Series.median.html*

### <b><span style='color:#55679C'>Quest C.2</span></b>
<i>Pada dataframe `carbon_product`, hitung median carbon footprint yang dihasilkan oleh semua produk 'iPhone'!</i>

In [28]:
# Write your code below
median_carbon_footprint = carbon_product['Carbon Footprint'].median()

print(f"Median carbon footprint untuk produk yang mengandung 'iPhone': {median_carbon_footprint}")

Median carbon footprint untuk produk yang mengandung 'iPhone': 62.5


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

---
## <b>C.3. <span style='color:#3A6D8C'>Modus (Mod)</span></b>
---

Modus adalah data yang paling sering muncul atau frekuensi paling tinggi. Modus dapat memberikan perkiraan nilai yang paling umum atau sering muncul dalam sekelompok data. <br><br>

Contoh :     

Kuantitas pembelanjaan barang pada sebuah toko di tanggal tertentu dicatat sebagai berikut :

<table align="center">
  <tr>
    <th>Trx ID</th>
    <th>Quantity</th>
    <th>Price</th>
  </tr>
  <tr>
    <td align="center">TRX001</td>
    <td align="center">4</td>
    <td align="center">55.000</td>
  </tr>
  <tr>
    <td align="center">TRX002</td>
    <td align="center">1</td>
    <td align="center">20.000</td>
  </tr>
  <tr>
    <td align="center">TRX003</td>
    <td align="center">5</td>
    <td align="center">17.000</td>
  </tr>
  <tr>
    <td align="center">TRX004</td>
    <td align="center">10</td>
    <td align="center">7500</td>
  </tr>
  <tr>
    <td align="center">TRX005</td>
    <td align="center">1</td>
    <td align="center">189.000</td>
  </tr>
  <tr>
    <td align="center">TRX006</td>
    <td align="center">1</td>
    <td align="center">203.000</td>
  </tr>
  <tr>
    <td align="center">TRX007</td>
    <td align="center">4</td>
    <td align="center">95.000</td>
  </tr>
</table>

Berapa modus dari kuantitas transaksi?

**Jawab :**

Hitung masing - masing frekuensi setiap nilai <br><br>
<table align="center">
  <tr>
    <th>Quantity</th>
    <th>Frequency</th>
  </tr>
  <tr>
    <th>1</th>
    <th>3</th>
  </tr>
  <tr>
    <th>4</th>
    <th>2</th>
  </tr>
  <tr>
    <th>5</th>
    <th>1</th>
  </tr>
  <tr>
    <th>10</th>
    <th>1</th>
  </tr>
</table>

Maka diperoleh Modus Quantity = 1

> **Interpretasi** : *Pelanggan paling sering hanya membeli 1 barang tiap transaksi. Perusahaan perlu meninjau persediaan barang, mengadakan promo bundling, analisa trend, dll untuk meningkatkan penjualan dari sisi kuantitas*

Untuk menghitung modus dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.mode()
```

docs : *https://pandas.pydata.org/docs/reference/api/pandas.Series.mode.html*

### <b><span style='color:#55679C'>Quest C.3</span></b>
<i>Pada dataframe `carbon_product`, hitung modus carbon footprint yang dihasilkan oleh semua produk 'iPhone'!</i>

In [34]:
carbon_product.sort_values(by = 'Carbon Footprint')

Release Year    Product  Baseline Storage  Carbon Footprint
9          2015  iPhone 6s                32                54
0          2023  iPhone 15               128                56
8          2016   iPhone 7                32                56
7          2017   iPhone 8                64                57
1          2022  iPhone 14               128                61
2          2021  iPhone 13               128                64
3          2020  iPhone 12                64                70
5          2018  iPhone Xs                64                70
4          2019  iPhone 11                64                72
6          2017   iPhone X                64                79

In [30]:
modus = carbon_product['Carbon Footprint'].mode()[0]
print(f"Modus carbon footprint yang dihasilkan oleh iPhone adalah {modus}")

Modus carbon footprint yang dihasilkan oleh iPhone adalah 56


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

---
## <b>C.4. <span style='color:#3A6D8C'>Kuartil (<i>Q</i>)</span></b>
---

Kuartil adalah nilai-nilai yang membagi data menjadi empat bagian yang sama. Ada tiga kuartil dalam sebuah data:

* Kuartil pertama (Q1) yang membagi data menjadi 25% terendah
* Kuartil kedua (Q2) yang merupakan median dan membagi data menjadi dua bagian yang sama
* Kuartil ketiga (Q3) yang membagi data menjadi 75% terendah atau 25% teratas <br><br>

<p align="center">
<img src="https://www.scribbr.com/wp-content/uploads/2022/05/Quartile-example.webp" width="500">
<p align="center"><b>Gambar 2. Kuartil</b></p>

Contoh :     

Kuantitas pembelanjaan barang pada sebuah toko di tanggal tertentu dicatat sebagai berikut :

<table align="center">
  <tr>
    <th>Trx ID</th>
    <th>Quantity</th>
    <th>Price</th>
  </tr>
  <tr>
    <td align="center">TRX001</td>
    <td align="center">4</td>
    <td align="center">55.000</td>
  </tr>
  <tr>
    <td align="center">TRX002</td>
    <td align="center">1</td>
    <td align="center">20.000</td>
  </tr>
  <tr>
    <td align="center">TRX003</td>
    <td align="center">5</td>
    <td align="center">17.000</td>
  </tr>
  <tr>
    <td align="center">TRX004</td>
    <td align="center">10</td>
    <td align="center">7500</td>
  </tr>
  <tr>
    <td align="center">TRX005</td>
    <td align="center">1</td>
    <td align="center">189.000</td>
  </tr>
  <tr>
    <td align="center">TRX006</td>
    <td align="center">1</td>
    <td align="center">203.000</td>
  </tr>
  <tr>
    <td align="center">TRX007</td>
    <td align="center">4</td>
    <td align="center">95.000</td>
  </tr>
</table>

Hitung kuartil dari harga barang terjual!


**Jawab :**

Urutkan data terlebih dahulu
<table align="center">
  <tr>
    <th>7.500</th>
    <th>17.000</th>
    <th>20.000</th>
    <th>55.000</th>
    <th>95.000</th>
    <th>189.000</th>
    <th>203.000</th>
  </tr>
</table>

Diperoleh
* Q1 = Rp. 17.000
* Q2 = Rp. 55.000
* Q3 = Rp. 189.000

> **Interpretasi** : *Sebanyak 25% barang yang terjual harganya dibawah Rp. 17.000 dan sebanyak 25% barang yang terjual harganya diatas Rp. 189.000 dengan nilai tengah yang diperoleh Rp. 55.000*


Untuk menghitung kuartil dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.quantile(q)
```

Untuk
* Q1 maka q = 0.25
* Q2 maka q = 0.50
* Q3 maka q = 0.75

docs : *https://pandas.pydata.org/docs/reference/api/pandas.Series.quantile.html*

### <b><span style='color:#55679C'>Quest C.4</span></b>
<i>Pada dataframe `normalizing_factors`, hitung semua kuartil untuk Kapitalisasi Pasar Apple dan berikan interpretasinya!</i>

In [35]:
normalizing_factors

Fiscal Year  Revenue  Market Capitalization  Employees
0         2022   394328                   2490     164000
1         2021   365817                   2450     154000
2         2020   274515                   1720     147000
3         2019   260174                   1090     137000
4         2018   265595                    830     132000
5         2017   229234                    740     123000
6         2016   215639                    600     116000
7         2015   233715                    580     110000

In [33]:
#Write your code below
quartiles = normalizing_factors['Market Capitalization'].quantile([0.25, 0.5, 0.75])
print(f"Kuartil kapitalisasi pasar Apple:\nKuartil (Q1): {quartiles[0.25]}\nKuartil (Q2): {quartiles[0.5]}\nKuartil (Q3): {quartiles[0.75]}")

Kuartil kapitalisasi pasar Apple:
Kuartil (Q1): 705.0
Kuartil (Q2): 960.0
Kuartil (Q3): 1902.5


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

# <b>D. <span style='color:#3A6D8C'>Dispersion</span></b>

Dispersion (penyebaran) dalam statistika merujuk pada ukuran variabilitas atau sebaran data. Hal ini menggambarkan sejauh mana nilai-nilai data tersebar di sekitar pusat distribusi

Ukuran penyebaran yang umum digunakan adalah :
* Range (rentang)
* Simpangan Baku dan Varians
* Interquartile Range (Selisih kuartil 1 dan 3)


---
## <b>D.1. <span style='color:#3A6D8C'>Range</span></b>
---

Rentang (range) adalah ukuran statistik yang menggambarkan selisih antara nilai maksimum dan nilai minimum dalam suatu set data. Dalam konteks statistika, rentang memberikan gambaran kasar tentang sebaran data, yaitu sejauh mana nilai-nilai data tersebar di antara nilai maksimum dan minimum.<br><br>

\begin{equation}
Range =  Max\;(Data)- Min\;(Data)
\end{equation}
<br>

Contoh :      

Gaji 5 orang pegawai IT di PT. X dengan posisi masing-masingnya adalah sebagai berikut :

<table align="center">
  <tr>
    <th>Nomor</th>
    <th align="left">Posisi</th>
    <th>Gaji</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="left">Data Analyst</td>
    <td align="center">5.500.000</td>
  </tr>
  <tr>
    <td align="center">2</td>
    <td align="left">Data Scientist</td>
    <td align="center">7.500.000</td>
  </tr>
  <tr>
    <td align="center">3</td>
    <td align="left">Data Engineer</td>
    <td align="center">8.000.000</td>
  </tr>
  <tr>
    <td align="center">4</td>
    <td align="left">Business Intelligence</td>
    <td align="center">7.000.000</td>
  </tr>
  <tr>
    <td align="center">5</td>
    <td align="left">Quality Assurance</td>
    <td align="center">5.000.000</td>
  </tr>
</table>
<br>

Berapa range gaji karyawan IT di perusahaan tersebut?

**Jawab :**

Gaji tertinggi adalah Data Scientist sebesar Rp. 8.000.000 sedangkan gaji paling rendah adalah Quality Assurance sebesar Rp. 5000.000 maka

Range = 8.000.000 - 5.000.000<br>
Range = 3.000.000

> **Interpretasi** : *Nilai-nilai data gaji tersebar di antara nilai maksimum dan minimum sebesar Rp.3000.000*

Untuk menghitung range dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.max() - Series.min()
```

docs 1 : *https://pandas.pydata.org/docs/reference/api/pandas.Series.max.html* <br>
docs 2 : *https://pandas.pydata.org/docs/reference/api/pandas.Series.min.html*

### <b><span style='color:#55679C'>Quest D.1</span></b>
<i>Pada dataframe `normalizing_factors`, hitung range Kapitalisasi Pasar Apple dan berikan interpretasinya!</i>

In [36]:
normalizing_factors

Fiscal Year  Revenue  Market Capitalization  Employees
0         2022   394328                   2490     164000
1         2021   365817                   2450     154000
2         2020   274515                   1720     147000
3         2019   260174                   1090     137000
4         2018   265595                    830     132000
5         2017   229234                    740     123000
6         2016   215639                    600     116000
7         2015   233715                    580     110000

In [32]:
#Write your code below
max_market_cap = normalizing_factors['Market Capitalization'].max()
min_market_cap = normalizing_factors['Market Capitalization'].min()

range_market_cap = max_market_cap - min_market_cap
print(f"Kapitalisasi Pasar maksimum: {max_market_cap}")
print(f"Kapitalisasi Pasar minimum: {min_market_cap}")
print(f"Range Kapitalisasi Pasar: {range_market_cap}")

Kapitalisasi Pasar maksimum: 2490
Kapitalisasi Pasar minimum: 580
Range Kapitalisasi Pasar: 1910


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
Range yang besar menunjukkan bahwa market capitalization apple bervariasi selama beberapa tahun terakhir. misal jika nilai maksimum jauh lebih tinggi dibanding nilai minimum, maka bisa menunjukkan bahwa si apple mengalami pertumbuhan besar dalam market capitalization di periode waktu 2015-2022

---
## <b>D.2. <span style='color:#3A6D8C'>Standard Deviation & Variance</span></b>
---

Simpangan baku (standard deviation) adalah ukuran statistik yang digunakan untuk mengukur sebaran atau dispersi data dalam sebuah sampel atau populasi. Simpangan baku menggambarkan sejauh mana nilai-nilai data tersebar di sekitar rata-rata (mean) dalam suatu distribusi.

\begin{equation}
\sigma = \sqrt{\frac{\sum{(x - \mu)^2}}{N}}
\end{equation}

* σ adalah simpangan baku.
* 𝒙 adalah nilai data dalam sampel atau populasi.
* μ adalah rata-rata (mean) dari data.
* ∑ adalah simbol sigma yang menunjukkan penjumlahan.
* 𝑵 adalah jumlah total nilai dalam sampel atau populasi.
<br><br>

Sedangkan variansi adalah kuadrat dari simpangan baku atau

\begin{equation}
var = \sigma^2
\end{equation}
<br>

Contoh :      

Gaji 5 orang pegawai IT di PT. X dengan posisi masing-masingnya adalah sebagai berikut :

<table align="center">
  <tr>
    <th>Nomor</th>
    <th align="left">Posisi</th>
    <th>Gaji</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="left">Data Analyst</td>
    <td align="center">5.500.000</td>
  </tr>
  <tr>
    <td align="center">2</td>
    <td align="left">Data Scientist</td>
    <td align="center">7.500.000</td>
  </tr>
  <tr>
    <td align="center">3</td>
    <td align="left">Data Engineer</td>
    <td align="center">8.000.000</td>
  </tr>
  <tr>
    <td align="center">4</td>
    <td align="left">Business Intelligence</td>
    <td align="center">7.000.000</td>
  </tr>
  <tr>
    <td align="center">5</td>
    <td align="left">Quality Assurance</td>
    <td align="center">5.000.000</td>
  </tr>
</table>
<br>

Berapa simpangan baku gaji karyawan IT di perusahaan tersebut?

**Jawab :**

Diketahui rata - rata dari data tersebut adalah μ = Rp. 6.600.000, maka

<table align="center">
  <tr>
    <th>Nomor</th>
    <th align="center">𝒙</th>
    <th>𝒙 - μ</th>
    <th>(𝒙 - μ)^2</th>
  </tr>
  <tr>
    <td align="center">1</td>
    <td align="center">5.500.000</td>
    <td align="center">-1.100.000</td>
    <td align="center">1,21 T</td>
  </tr>
  <tr>
    <td align="center">2</td>
    <td align="center">7.500.000</td>
    <td align="center">900.000</td>
    <td align="center">0,81 T</td>
  </tr>
  <tr>
    <td align="center">3</td>
    <td align="center">8.000.000</td>
    <td align="center">1.400.000</td>
    <td align="center">1,96 T</td>
  </tr>
  <tr>
    <td align="center">4</td>
    <td align="center">7.000.000</td>
    <td align="center">400.000</td>
    <td align="center">0,16 T</td>
  </tr>
  <tr>
    <td align="center">5</td>
    <td align="center">5.000.000</td>
    <td align="center">-1.600.000</td>
    <td align="center">2,56 T</td>
  </tr>
  <tr>
    <td align="center"></td>
    <td align="center"></td>
    <td align="center">Total</td>
    <td align="center">6,70 T</td>
  </tr>
</table>
<br>

\begin{equation}
\sigma = \sqrt{\frac{\sum{(x - \mu)^2}}{N}} = \sqrt{\frac{6,70\;T}{5}} = 1.157.584
\end{equation}
<br>

> **Interpretasi** : *Simpangan baku yang relatif sedang (Rp. 1.157.584) menunjukkan variasi gaji yang cukup signifikan di antara individu-individu dalam populasi yang sedang dianalisis. Ini menunjukkan bahwa ada perbedaan yang cenderung besar antara gaji tertinggi dan terendah dalam populasi tersebut*

Untuk menghitung simpangan baku dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.std()
```

dan variansi

```
Series.var()
```

docs 1 : *https://pandas.pydata.org/docs/reference/api/pandas.Series.std.html* <br>
docs 2 : *https://pandas.pydata.org/docs/reference/api/pandas.Series.var.html*

### <b><span style='color:#55679C'>Quest D.2</span></b>
<i>Pada dataframe `gas_emission`, hitung simpangan baku gas emisi penggunaan produk Apple dan berikan interpretasinya!</i>

In [37]:
gas_emissions['Description'].unique().tolist()

['Natural gas, diesel, propane',
 'Fleet vehicles',
 'Other (R&D processes & refrigerant leaks)',
 'Electricity',
 'Steam, heating, and cooling',
 'Business travel',
 'Employee commute',
 'Upstream fuel',
 'Work from home (market-based)',
 'Transmission and distribution loss (market-based)',
 'Third-party cloud (market-based)',
 'Corporate carbon offsets',
 'Manufacturing (purchased goods and services)',
 'Product transportation (upstream and downstream)',
 'Product use (use of sold products)',
 'End-of-life processing',
 'Product carbon offsets']

In [38]:
condition = gas_emissions['Description'] == 'Product use (use of sold products)'
product_use = gas_emissions.loc[condition]
product_use

Fiscal Year                      Category             Type    Scope  \
14          2022  Product life cycle emissions  Gross emissions  Scope 3   
31          2021  Product life cycle emissions  Gross emissions  Scope 3   
48          2020  Product life cycle emissions  Gross emissions  Scope 3   
65          2019  Product life cycle emissions  Gross emissions  Scope 3   
82          2018  Product life cycle emissions  Gross emissions  Scope 3   
99          2017  Product life cycle emissions  Gross emissions  Scope 3   
116         2016  Product life cycle emissions  Gross emissions  Scope 3   
133         2015  Product life cycle emissions  Gross emissions  Scope 3   

                            Description  Emissions  
14   Product use (use of sold products)  4900000.0  
31   Product use (use of sold products)  4990000.0  
48   Product use (use of sold products)  4300000.0  
65   Product use (use of sold products)  4100000.0  
82   Product use (use of sold products)  4700000.0  
99   Product use (use of sold products)  4700000.0  
116  Product use (use of sold products)  4900000.0  
133  Product use (use of sold products)  6600000.0

In [39]:
product_use['Emissions'].std(ddof=0)

705167.3117069452

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

---
## <b>D.3. <span style='color:#3A6D8C'>Interquartile Range</span></b>
---

Interquartile Range (IQR) adalah ukuran statistik yang digunakan untuk mengukur sebaran data dalam sebuah sampel atau populasi, terutama dalam konteks analisis data yang melibatkan kuartil.

\begin{equation}
IQR = Q_3 - Q_1
\end{equation}

 IQR dapat memberikan petunjuk tentang bentuk distribusi data. Jika IQR relatif sempit (kecil nilainya), artinya data cenderung terkonsentrasi di sekitar median dengan sedikit variasi. Namun, jika IQR relatif lebar, artinya data lebih terdispersi dan dapat menunjukkan variasi yang signifikan dalam distribusi.

 Contoh :     

Kuantitas pembelanjaan barang pada sebuah toko di tanggal tertentu dicatat sebagai berikut :

<table align="center">
  <tr>
    <th>Trx ID</th>
    <th>Quantity</th>
    <th>Price</th>
  </tr>
  <tr>
    <td align="center">TRX001</td>
    <td align="center">4</td>
    <td align="center">55.000</td>
  </tr>
  <tr>
    <td align="center">TRX002</td>
    <td align="center">1</td>
    <td align="center">20.000</td>
  </tr>
  <tr>
    <td align="center">TRX003</td>
    <td align="center">5</td>
    <td align="center">17.000</td>
  </tr>
  <tr>
    <td align="center">TRX004</td>
    <td align="center">10</td>
    <td align="center">7500</td>
  </tr>
  <tr>
    <td align="center">TRX005</td>
    <td align="center">1</td>
    <td align="center">189.000</td>
  </tr>
  <tr>
    <td align="center">TRX006</td>
    <td align="center">1</td>
    <td align="center">203.000</td>
  </tr>
  <tr>
    <td align="center">TRX007</td>
    <td align="center">4</td>
    <td align="center">95.000</td>
  </tr>
</table>

Hitung kuartil dari harga barang terjual!


**Jawab :**

Urutkan data terlebih dahulu
<table align="center">
  <tr>
    <th>7.500</th>
    <th>17.000</th>
    <th>20.000</th>
    <th>55.000</th>
    <th>95.000</th>
    <th>189.000</th>
    <th>203.000</th>
  </tr>
</table>

Diperoleh
* Q1 = Rp. 17.000
* Q2 = Rp. 55.000
* Q3 = Rp. 189.000
<br>

Maka IQR = Q3 - Q1 = 189.000 - 17.000 = 172.000
<br>

> **Interpretasi** : *Diperoleh IQR cukup besar Rp. 172.000 artinya data lebih terdispersi dan dapat menunjukkan variasi yang signifikan dalam distribusi*


Untuk menghitung IQR dari sebuah kolom pada DataFrame dapat menggunakan sintaks

```
Series.quantile(q = 0.75) - Series.quantile(q = 0.25)
```

docs : *https://pandas.pydata.org/docs/reference/api/pandas.Series.quantile.html*

### <b><span style='color:#55679C'>Quest D.3</span></b>
<i>Pada dataframe `gas_emission`, hitung IQR gas emisi penggunaan produk Apple dan berikan interpretasinya!</i>

In [40]:
# Write your code below!
product_use_data = gas_emissions[gas_emissions['Description'] == 'Product use (use of sold products)']

Q1 = product_use_data['Emissions'].quantile(0.25)
Q2 = product_use_data['Emissions'].quantile(0.5)
Q3 = product_use_data['Emissions'].quantile(0.75)
IQR = Q3 - Q1

print(f'Q1 emisi gas kategori Product use: {Q1}')
print(f'Q2 (Median) emisi gas kategori Product use: {Q2}')
print(f'Q3 emisi gas kategori Product use: {Q3}')
print(f'IQR emisi gas kategori Product use: {IQR}')

Q1 emisi gas kategori Product use: 4600000.0
Q2 (Median) emisi gas kategori Product use: 4800000.0
Q3 emisi gas kategori Product use: 4922500.0
IQR emisi gas kategori Product use: 322500.0


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

# <b>E. <span style='color:#3A6D8C'>Outliers</span></b>

Outliers (pencilan) adalah nilai-nilai yang sangat berbeda atau jauh dari nilai-nilai lain dalam suatu dataset. Mereka adalah observasi yang ekstrem atau tidak biasa dibandingkan dengan nilai-nilai lainnya. Outliers bisa menjadi hasil dari kesalahan pengukuran, perbedaan dalam populasi atau sebenarnya, atau mungkin merupakan indikasi dari variasi yang signifikan dalam data.

Outliers dapat diidentifikasi menggunakan metode statistik, seperti menggunakan batas atas dan batas bawah berdasarkan kriteria tertentu, atau dengan memvisualisasikan data menggunakan grafik seperti box plot atau scatter plot. Pendekatan yang umum digunakan adalah dengan menggunakan metode seperti kuartil (interquartile range/IQR) atau tiga kali deviasi standar.
<br><br>

\begin{equation}
\text{Lower Fence} = Q_1 - 1.5 * IQR\\
\text{Upper Fence} = Q_3 + 1.5 * IQR
\end{equation}
<br>

Jika sebuah nilai pada data dibawah nilai Lower Fence disebut *low outlier* sedangkan nilai data diatas Upper Fence disebut *high outlier* <br><br>

<p align="center">
<img src="https://miro.medium.com/v2/resize:fit:9000/1*2c21SkzJMf3frPXPAR_gZA.png" width="550" height="275">
<p align="center"><b>Gambar 3. Box-Plot</b></p>
</p>


docs : *https://plotly.com/python/box-plots/*

**Perhatikan Contoh Berikut :**

Pada Python, akan dibuat outlier labeling pada sebuah data yakni dengan kriteria

* Jika nilai dibawah Lower Hence maka beri label 'Low Outlier'
* Jika nilai diatas Upper Hence maka beri label 'High Outlier'
* Selain itu beri label 'InLier' (lawan kata outlier)

In [41]:
# Proses penentuan ambang batas outlier

def hitung_batas_outlier(data):
    Q1 = data.quantile(q = 0.25)
    Q3 = data.quantile(q = 0.75)
    IQR = Q3 - Q1
    lower_hence = Q1 - 1.5 * IQR
    upper_hence = Q3 + 1.5 * IQR
    return(lower_hence, upper_hence)

In [42]:
# Proses pemberian label outlier pada data

def beri_label_outlier(nilai, lower_fence, upper_fence):
    if(nilai < lower_fence):
        tipe = 'Low Outlier'
    elif(nilai > upper_fence):
        tipe = 'High Outlier'
    else:
        tipe = 'InLier'

    return(tipe)

In [ ]:
# Panggil fungsi untuk menghitung batas outlier
lower_fence, upper_fence = hitung_batas_outlier(data = gas_emissions['Emissions'])

# Tampilkan hasilnya
print(f'Lower Hence pada kolom Emissions ={lower_fence : .2f}')
print(f'Upper Hence pada kolom Emissions ={upper_fence : .2f}')

Lower Hence pada kolom Emissions =-450000.00
Upper Hence pada kolom Emissions = 750000.00


In [ ]:
# Setelah lower_hence dan upper_hence ditemukan,
gas_emissions['Type Outlier Emissions'] = gas_emissions['Emissions'].apply(beri_label_outlier, args = (lower_fence, upper_fence))

# Tampilkan hasilnya
display(gas_emissions[['Emissions', 'Type Outlier Emissions']])

Emissions Type Outlier Emissions
0       39700.0                 InLier
1       12600.0                 InLier
2        2900.0                 InLier
3           0.0                 InLier
4        3000.0                 InLier
..          ...                    ...
131  29600000.0           High Outlier
132   1300000.0           High Outlier
133   6600000.0           High Outlier
134    500000.0                 InLier
135         NaN                 InLier

[136 rows x 2 columns]

### <b><span style='color:#55679C'>Quest E.1</span></b>
<i>Pada dataframe `gas_emission`, telah ditambahkan kolom `Type Outlier Emissions` sekarang lakukan pengelompokan masing - masing tipe dan hitung ada berapa banyak data yang masuk kategori tersebut!</i>

In [ ]:
# Mengelompokkan data berdasarkan nilai kolom Type Outlier Profit
gas_emissions.groupby('Type Outlier Emissions', as_index=False).agg(Total = ('Type Outlier Emissions', 'count'))

Type Outlier Emissions  Total
0           High Outlier     24
1                 InLier    111
2            Low Outlier      1

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
* High Outlier (24 data) menunjukkan adanya emisi yang sangat tinggi.
* InLier (111 data) adalah data yang berada dalam kisaran normal dan merupakan mayoritas dari dataset.
* Low Outlier (1 data) mungkin merepresentasikan emisi yang sangat rendah.

### <b><span style='color:#55679C'>Quest E.2</span></b>
<i>Visualisasi dengan Box-Plot akan mempermudah deteksi outlier dari sebuah data. Dengan dataset `gas_emission`, coba visualisasikan kolom Emissions menggunakan Box-Plot untuk melihat outlier dari data!</i>

Note : Kalian bisa menggunakan referensi berikut untuk melakukan visualisasi data dengan boxplot<br>
docs : *https://plotly.com/python/box-plots/#box-plot-with-plotlyexpress*


<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
Kotak (Box):
* Bagian kotak menunjukkan rentang interkuartil (IQR), yaitu bagian tengah 50% dari data.
Garis di tengah kotak (median) menggambarkan nilai tengah dari distribusi emisi.

Whiskers:
* Garis horizontal yang keluar dari kotak (whiskers) menunjukkan rentang data di luar IQR, biasanya hingga 1,5 kali IQR dari kuartil pertama (Q1) dan kuartil ketiga (Q3). Whiskers memberikan gambaran tentang sebaran data dan menunjukkan nilai-nilai yang masih dianggap normal.

Outlier:
* Titik di luar whiskers biasanya dianggap sebagai outlier, yaitu data yang jauh lebih tinggi atau lebih rendah daripada nilai lainnya. Outlier bisa menunjukkan anomali dalam data, seperti emisi yang sangat tinggi atau rendah, yang mungkin memerlukan perhatian lebih lanjut.

Distribusi Data:
* Dari box plot, kita bisa melihat apakah distribusi data simetris atau miring. Jika median lebih dekat ke batas bawah kotak, berarti distribusi miring ke kanan (positif), sedangkan jika lebih dekat ke batas atas, berarti miring ke kiri (negatif).
Singkatnya :
Box plot memberikan informasi penting tentang sebaran emisi gas, membantu untuk:
* Mengidentifikasi seberapa besar emisi secara keseluruhan.
* Mengetahui adanya outlier yang perlu diselidiki lebih lanjut.
* Memahami distribusi dan variasi data emisi dalam dataset Anda.
Jadi hanya untuk membuat keputusan yang lebih tepat dalam konteks pengelolaan emisi gas rumah kaca.

In [43]:
# Write your code below!
import plotly.express as px

fig = px.box(
    gas_emissions,
    x = 'Emissions',
    title = 'Box Plot of Emissions'
)

# Fig = Figure
fig.show()

### <b><span style='color:#55679C'>Quest E.3</span></b>
*Coba cari dibeberapa referensi bacaan, apa dampak outlier pada data (terutama pengaruhnya terhadap statistik deskriptif data)? Lalu coba interpretasikan analisamu dari hasil visualisasi Box-Plot Emissions diatas dan jelaskan!*

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
Dampak Outliers pada data :
1. Rata-rata
   Outlier memiliki dampak signifikan terhadap rata-rata. Jika outlier sangat besar atau sangat kecil dibandingkan dengan nilai lain, maka rata-rata akan bergerak ke arah outlier tersebut. Misalnya, jika sebagian besar data berada di kisaran 10-20, tetapi terdapat satu nilai sebesar 100, maka rata-rata akan meningkat signifikan.

2. Median dan Modus
   Median dan modus relatif lebih stabil terhadap outlier karena tidak dipengaruhi oleh besarnya nilai individu. Median hanya melihat nilai tengah, sedangkan modus hanya melihat frekuensi kemunculan. Namun, jika outlier jumlahnya banyak, median bisa sedikit bergeser.

3. Range
   Outlier dapat meningkatkan rentang secara signifikan karena rentang dihitung dari nilai minimum ke nilai maksimum. Kehadiran outlier dengan nilai ekstrem akan memperlebar rentang data, membuat variabilitas terlihat lebih besar daripada yang sebenarnya.

4. Standar Deviasi dan Varians
   Outlier akan meningkatkan standar deviasi dan varians karena kedua ukuran ini sensitif terhadap jarak tiap nilai data terhadap rata-rata. Ketika terdapat outlier, nilai jarak rata-rata menjadi lebih besar, sehingga standar deviasi dan varians meningkat.

5. Skewness dan Kurtosis
   Outlier dapat memengaruhi skewness (kemiringan) dan kurtosis (keruncingan) dari distribusi data. Outlier pada ujung-ujung distribusi bisa membuat data terlihat lebih miring atau lebih runcing dibandingkan dengan distribusi normal.

### <b><span style='color:#55679C'>Quest E.4</span></b>
*Coba cari dibeberapa referensi bacaan, bagaimana cara handling outlier (mengatasi outlier) pada data dan jelaskan!*

<b><span style='color:#FFDB00'><i>Penjelasan : </i></span></b><br>
Strategi untuk Menangani Outlier Pada dasarnya terdapat banyak cara untuk menangani outlier, dan metode yang tepat tergantung pada sifat data dan masalah spesifik yang diselesaikan. Namun, strategi umum dalam menangani outlier dirincikan sebagai berikut:
Deletion
Deletion (penghapusan) atau membuang adalah cara paling sederhana untuk menangani outlier, tetapi harus digunakan dengan hati-hati. Kita hanya perlu menghapus pencilan jika kita yakin bahwa pencilan tersebut disebabkan oleh data yang salah dimasukkan atau diukur. Sebab, menghapus titik data yang berharga dapat menyebabkan hilangnya informasi dan hasil yang bias.

Transformation
Mentransformasi variabel juga dapat membantu meminimalkan dampak outlier. Transformasi yang umum dilakukan meliputi transformasi log, akar kuadrat (square root), dan invers. Transformasi ini dapat memadatkan nilai yang lebih tinggi, sehingga mengurangi efek dari nilai ekstrem.

Winsorization
Pada aturan winsorization, nilai ekstrem atau outlier diganti dengan persentil tertentu (biasanya persentil ke-5 dan ke-95). Teknik ini bertujuan untuk mempertahankan ukuran dataset yang tidak diantisipasi pada metode deletion sebelumnya.

Pemilihan Model Machine Learning
Untuk mereduksi pengaruh keberadaan outlier pada data, salah satu teknik yang dapat dilakukan adalah dengan memilih model machine learning yang robust terhadap outlier. Beberapa model machine learning yang robust terhadap outlier adalah Support Vector Machine (SVM), Random Forest, dan Gradient Boosting. Cara kerja dari masing-masing model tersebut tidak tergantung atau tidak dipengaruhi oleh keberadaan outlier, sehingga penggunaannya dapat menjadi salah satu alternatif untuk mereduksi atau menghilangkan pengaruh keberadaan outlier pada data.

---

# <b><span style='color:#3A6D8C'>Notes</span></b>

Pada bagian F & G akan digunakan dataset `Data Temperature.csv` dengan penjelasan kolom :

* `date`: Kolom yang berisi tanggal atau timestamp untuk setiap pengamatan.
* `Iws (Instantaneous Wind Speed)`: Kolom yang berisi kecepatan angin sesaat pada saat pengamatan. Satuan kecepatan angin biasanya dalam meter per detik (m/s) atau knot.
* `Ir (Instantaneous Radiation)`: Kolom yang berisi intensitas radiasi matahari sesaat pada saat pengamatan. Satuan radiasi biasanya dalam watt per meter persegi (W/m²).
* `pm2.5`: Kolom yang berisi konsentrasi partikel halus (particulate matter) berukuran 2.5 mikrometer atau lebih kecil. Satuan biasanya dalam mikrogram per meter kubik (μg/m³).
* `PRES (Pressure)`: Kolom yang berisi tekanan udara pada saat pengamatan. Satuan tekanan biasanya dalam hektopascal (hPa) atau millibar (mb).
* `cbwd (Combined Wind Direction)`: Kolom yang berisi arah angin pada saat pengamatan. Data dalam kolom ini biasanya dalam bentuk kode atau kategori, misalnya N, S, E, W, NW, dll.
* `DEWP (Dew Point)`: Kolom yang berisi titik embun pada saat pengamatan. Titik embun adalah suhu di mana uap air di udara akan mulai mengembun. Satuan biasanya dalam derajat Celsius (°C).

Source : https://www.kaggle.com/competitions/probabilistic-forecasting-i-temperature/data?select=train.csv

In [ ]:
import pandas as pd

temperature = pd.read_csv('Data Temperature.csv')

display(temperature)

id                 date  feature_AA  feature_AB  feature_BA  \
0          0  2016-07-01 00:00:00       5.827       2.009       1.599   
1          1  2016-07-01 00:15:00       5.760       2.076       1.492   
2          2  2016-07-01 00:30:00       5.760       1.942       1.492   
3          3  2016-07-01 00:45:00       5.760       1.942       1.492   
4          4  2016-07-01 01:00:00       5.693       2.076       1.492   
...      ...                  ...         ...         ...         ...   
64315  64315  2018-05-01 22:45:00      12.458       1.407       7.889   
64316  64316  2018-05-01 23:00:00      14.200       1.541       9.310   
64317  64317  2018-05-01 23:15:00      12.659       1.474       7.924   
64318  64318  2018-05-01 23:30:00      13.329       1.541       8.671   
64319  64319  2018-05-01 23:45:00      11.588       0.871       7.214   

       feature_BB  feature_CA  feature_CB  Temperature  
0           0.462       4.203       1.340    30.531000  
1           0.426       4.264       1.401    30.459999  
2           0.391       4.234       1.310    30.038000  
3           0.426       4.234       1.310    27.013000  
4           0.426       4.142       1.371    27.787001  
...           ...         ...         ...          ...  
64315      -0.746       4.843       1.523    10.622000  
64316       0.284       4.752       1.249    10.482000  
64317      -0.959       4.569       1.310    10.271000  
64318      -0.533       4.630       1.371    10.341000  
64319      -0.426       4.691       1.310    10.833000  

[64320 rows x 9 columns]

---

# <b>F. <span style='color:#3A6D8C'>Distribusi</span></b>

Dalam konteks analisis data, distribusi menggambarkan pola atau karakteristik bagaimana data tersebar.

Tujuan melihat distribusi data :

* Identifikasi Pusat Data: Distribusi membantu mengidentifikasi nilai tengah data, seperti mean, median, dan modus.
* Deteksi Pencilan (Outliers): Distribusi membantu mendeteksi adanya pencilan dalam data.
* Normalitas: Distribusi digunakan untuk mengevaluasi apakah data mengikuti distribusi normal atau tidak.
* Pemilihan Model: Distribusi membantu memilih model yang tepat untuk menganalisis data.
* Perhitungan Statistik Deskriptif: Distribusi penting untuk menghitung statistik deskriptif seperti deviasi standar, kuartil, dan persentil.<br><br>

<p align="center">
<img src="https://i.stack.imgur.com/Qfc12.png">
<p align="center"><b>Gambar 1. Distribution</b></p>
</p>




### <b><span style='color:#55679C'>Quest F.1</span></b>
*Dari dataset `temperature`, coba buat histogram pada kolom Emissions dan perhatikan persebaran datanya!*

<ol>
Note : Kalian bisa menggunakan referensi berikut untuk melakukan visualisasi data dengan histogram

docs : *https://plotly.com/python/histograms/#histograms-with-plotly-express*
</ol>

In [ ]:
# Histogram Sales
import plotly.express as px

fig = px.histogram(
    gas_emissions[gas_emissions['Type'] == 'Carbon removals'],
    x = 'Emissions',
    marginal = 'box',
    color_discrete_sequence  = ['#0E2954'],
    nbins = 20
)

fig.update_traces(
      marker_line_width = 1,
      marker_line_color = 'white'
)

fig.update_layout(
    plot_bgcolor = 'rgba(0, 0, 0, 0)',
    title = dict(
        text = "<b>Distribusi <span style='color:#0E2954'>Carbon Footprint</b> Produk iPhone<br>",
        font = dict(
            size = 28,
            color = '#757882'
        ),
        y = 0.92,
        x = 0.5
    ),
    yaxis = dict(
        title = '',
        showgrid = False,
        showline = False,
        showticklabels = False,
        zeroline = False,
    ),
    margin = dict(
        t = 80,
        b = 10,
        r = 20
    )
)

fig.show()

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

# <b>G. <span style='color:#3A6D8C'>Skewness</span></b>

Kemiringan (skewness) adalah sebuah ukuran statistik yang menggambarkan asimetri atau ketidaksimetrisan dari distribusi data. Ini memberikan informasi tentang sejauh mana ekor distribusi data miring ke kiri atau ke kanan relatif terhadap pusat distribusi.

\begin{equation}
\text{Skewness} = \frac{\sum_{i=1}^{N}(x_i - \bar{x})^3}{N \cdot \sigma^3}
\end{equation}
<br>

Jika
* Skewness < 0 maka disebut Negatively Skewed
* Skewness = 0 maka disebut Normal
* Skewness > 0 maka Positively Skewed
<br><br>

<p align="center">
<img src="https://www.biologyforlife.com/uploads/2/2/3/9/22392738/c101b0da6ea1a0dab31f80d9963b0368_orig.png" width="750" height="300">
<p align="center"><b>Gambar 1. Distribution</b></p>
</p>


### **Case 12** : *Dari hasil histogram sebelumnya, Coba pertimbangkan jenis Skewness masing-masing indikator (Sales, Quantity, Profit) lalu berikan interpretasinya!*

<b><span style='color:#FFDB00'><i>Penjelasan : </i></span></b><br>
...

# <b>H. <span style='color:#3A6D8C'>Korelasi</span></b>

Korelasi adalah sebuah konsep statistik yang mengukur sejauh mana dua variabel berkaitan satu sama lain. Korelasi menggambarkan hubungan linier antara dua variabel, yaitu sejauh mana perubahan dalam satu variabel berhubungan dengan perubahan dalam variabel lainnya. Dua variabel dikatakan berkorelasi apabila perubahan salah satu variabel disertai dengan perubahan variabel lainnya, baik dalam arah yang sama ataupun arah yang sebaliknya.
<br><br>

<p align="center">
<img src="https://www.mathsisfun.com/data/images/correlation-examples.svg">
<p align="center"><b>Gambar 2. Korelasi</b></p>

Untuk dapat menggambarkan kekuatan dan arah hubungan linier antara dua variabel dapat menggunakan koefisien korelasi yang nilainya berkisar antara -1 <= r <= 1.

<p align="center">
<img src="https://www.researchgate.net/profile/Fermin-Rodriguez-3/publication/344691053/figure/fig2/AS:947264293974017@1602856815816/Pearson-correlation-scale.png" width="600" height="151">
<p align="center"><b>Gambar 3. Skala Korelasi</b></p><br>

Penting untuk dicatat bahwa korelasi tidak menunjukkan hubungan sebab-akibat antara variabel. Korelasi hanya menggambarkan hubungan linier antara variabel-variabel tersebut.

## **8.1 Korelasi Pearson**

Sebuah metode untuk mengukur hubungan linier antara dua variabel interval numerik. Untuk menghitung koefisien korelasi Pearson dapat menggunakan sintaks

<center><code>DataFrame.corr(method = 'pearson')</code></center>


docs : *https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.corr.html*

### **Case 13** : *Dari dataset Super Store, hitung korelasi tiap indikator numerik pada data lalu periksa variabel apa yang saling berkorelasi dan berikan interpretasinya!*

In [ ]:
# Write your code!


Sales  Quantity  Discount    Profit  Shipping Cost
Sales          1.000000  0.555691 -0.187997  0.533676       0.644891
Quantity       0.555691  1.000000  0.034045  0.154481       0.356324
Discount      -0.187997  0.034045  1.000000 -0.505190      -0.169216
Profit         0.533676  0.154481 -0.505190  1.000000       0.290593
Shipping Cost  0.644891  0.356324 -0.169216  0.290593       1.000000

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

## **8.2 HeatMap**

Heatmap adalah visualisasi yang berguna untuk memvisualisasikan korelasi antara dua set data atau antara beberapa variabel dalam bentuk matriks.

docs : *https://plotly.com/python/heatmaps/*

In [ ]:
# Import library yang dibutuhkan
import plotly.express as px

# Hitung korelasi
korelasi = data_superstore[['Sales', 'Quantity', 'Discount', 'Profit', 'Shipping Cost']].corr()

# Buat Heat Map
fig = px.imshow(
    korelasi,
    text_auto = True,
    color_continuous_scale = 'Blues'
)

# Jangan tampilkan color axes / legend
fig.update_coloraxes(
    showscale = False
)

# Percantik tampilan
fig.update_layout(
    width = 800,
    height = 600,
    title = dict(
        text = "<b>Korelasi Antar Variable pada Data Super Store</b>",
        font = dict(
            size = 19,
            color = '#0E2954'
        ),
        y = 0.95,
        x = 0.5
    ),
    margin = dict(
        t = 80,
        b = 30,
        r = 50,
        l = 50
    )
)

# Tampilkan hasil visualisasi
fig.show()

In [ ]:
# Import library yang dibutuhkan
import plotly.express as px

# Buat scatter plot
fig = px.scatter(
    data_superstore,
    x = 'Sales',
    y = 'Shipping Cost',
    trendline = 'ols',
    color_discrete_sequence  = ['#0E2954']
)

# Tambahkan judul
fig.update_layout(
    title = dict(
        text = "<b>Korelasi Antar Variable Sales dan Shipping Cost<br>pada Data Super Store</b>",
        font = dict(
            size = 19,
            color = '#0E2954'
        ),
        y = 0.95,
        x = 0.5
    ),
)

# Tampilkan hasil visualisasinya
fig.show()

<b><span style='color:#FFDB00'><i>Interpretasi Hasil : </i></span></b><br>
...

---

<br>
<a href="https://www.linkedin.com/in/pandu-giswantoro-abhi-70a1a62a7/"><img src="https://img.shields.io/badge/-© 2024 Pandu Giswantoro Abhi Wiguno-417DAC?style=for-the-badge&logoColor=white"/></a>